pkg/resolution/resolver/cluster/resolver.go

```diff
@@ -122,8 +128,9 @@ func (r *Resolver) Resolve(ctx context.Context, origParams []pipelinev1beta1.Par
 			logger.Infof("failed to load pipeline %s from namespace %s: %v", params[NameParam], params[NamespaceParam], err)
 			return nil, err
 		}
+		uid = string(pipeline.UID)
 		pipeline.Kind = "Pipeline"
-		pipeline.APIVersion = "tekton.dev/v1beta1"
+		pipeline.APIVersion = groupVersion
```

Comment:
Out of curiosity, are we also going to support `v1` in the remote resolution after V1 release?

Reply:
Yeah, we will. Basically, this should always be whatever version is used in the `PipelineRun` reconciler by default, but we should probably add a param for `apiVersion` to allow overriding that...

Reply:
> Out of curiosity, are we also going to support `v1` in the remote resolution after V1 release?

Yeah agreed. We will need to support v1 in remote resolution. Thanks for asking @XinruZhang .

Reply:
+1 for a new param for apiVersion, it looks like a future refactor work XD